### ISSUE TITLE
too many open files

### ISSUE THREAD

**Issue by rbonthond** [amoffat/sh#358]
something like this causes an exception error

```
#!/usr/bin/env python3
import sh
for i in range(0, 10000):
    sh.echo('hi')
```

exception:
```
Traceback (most recent call last):
  File "./sh_pp.py", line 6, in <module>
    sh.echo('hi')
  File "/designtools/python3/lib/python3.5/site-packages/sh.py", line 1223, in __call__
    return RunningCommand(cmd, call_args, stdin, stdout, stderr)
  File "/designtools/python3/lib/python3.5/site-packages/sh.py", line 621, in __init__
    self.call_args, pipe, process_assign_lock)
  File "/designtools/python3/lib/python3.5/site-packages/sh.py", line 1614, in __init__
    exc_pipe_read, exc_pipe_write = os.pipe()
OSError: [Errno 24] Too many open files
```

Using Red Hat Enterprise Linux Server release 6.8 (Santiago)
linux cadaus004.silabs.com 2.6.32-642.6.2.el6.x86_64 #1 SMP Mon Oct 24 10:22:33 EDT 2016 x86_64 x86_64 x86_64 GNU/Linux

**Comment by amoffat** [comment]
@rbonthond what version of sh?  I am able to run your script fine on my system, and `ulimit -Hn` reports a hard limit of 4096 open file descriptors

**Comment by rbonthond** [comment]
I'm using sh 1.12.0, and I'm using tcsh
```
$ limit
cputime      unlimited
filesize     unlimited
datasize     unlimited
stacksize    10240 kbytes
coredumpsize 0 kbytes
memoryuse    unlimited
vmemoryuse   unlimited
descriptors  1024 
memorylocked 64 kbytes
maxproc      1024 
```

**Comment by amoffat** [comment]
Does the issue still persist after the upgrade to 1.12.9?

**Comment by rbonthond** [comment]
seems to be fixed since upgrading to 1.12.9, thank you!
